src/components/CustomInput.tsx

```diff
@@ -0,0 +1,43 @@
+import React, { FC } from 'react';
+import Form from 'react-bootstrap/Form';
+import { Col, Row } from 'react-bootstrap';
+
+interface CustomInputProps {
+  value: string,
+  controlId: string,
+  label: string,
+  placeholder?: string,
+  inputHandler: (val: React.ChangeEvent<HTMLInputElement>) => void,
+  valid: boolean,
+  type?: string,
+}
+
+const CustomInput: FC<CustomInputProps> = ({
+  controlId,
+  label,
+  placeholder,
+  inputHandler,
+  value,
+  valid,
+  type = 'text',
+}) => {
+  return (
+    <Form.Group as={Row} className="mb-3" controlId={controlId}>
+      <Form.Label column sm="2">
+        {label}
+      </Form.Label>
+      <Col sm="10">
+        <Form.Control
+          type={type}
+          placeholder={placeholder}
+          onChange={inputHandler}
+          value={value}
+          isValid={value.length > 1 && valid}
+          isInvalid={value.length > 1 && !valid}
```

Comment:
because the isValid or inValid properties should only be in customInput when it is not empty. If you check this in valid property, the input will be colored red or green by default. I tried to avoid this
![image](https://user-images.githubusercontent.com/41682520/133445873-5bbf11f0-d0ff-443c-89a6-9ae31f31258f.png)